src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -242,7 +243,12 @@ void CalorimeterHitReco::process(
         const auto pos = local.nominal().worldToLocal(gpos);
         std::vector<double> cdim;
         // get segmentation dimensions
-        auto segmentation_type = m_converter->findReadout(local).segmentation().type();
+        const Segmentation& segmentation = m_converter->findReadout(local).segmentation();
+        auto segmentation_type = segmentation.type();
+        while (segmentation_type == "MultiSegmentation"){
+            auto& multi_segmentation = dynamic_cast<DDSegmentation::MultiSegmentation&>(segmentation);
+            segmentation= multi_segmentation.subsegmentation(cellID);
+        }
```

Comment:
Yeah, this is running into the DD4hep handle structures. Try something like this:
```suggestion
        auto segmentation = m_converter->findReadout(local).segmentation();
        auto segmentation_type = segmentation.type();
        while (segmentation_type == "MultiSegmentation"){
            auto* multi_segmentation = dynamic_cast<dd4hep::DDSegmentation::MultiSegmentation*>(segmentation.segmentation());
            auto& sub_segmentation = multi_segmentation->subsegmentation(cellID);
            segmentation = dd4hep::Handle<dd4hep::DDSegmentation::Segmentation>(&sub_segmentation);
        }
```

Reply:
This compiles, but you'll want to test that it actually works.

Reply:
@veprbl may have thoughts too.

Reply:
I'm getting linker errors with this.  I'll try this locally after you get it to compile.

Reply:
I was able to get this to work, thanks to @nathanwbrei

Reply:
I see what you did there. Still curious how DD4hep would resolve this using the handles they impose on us :-)

Sorry, I shouldn't complain about handles...

https://github.com/AIDASoft/DD4hep/blob/9c6040cfa7918691b02024a7d161d9d340a5da30/DDCore/include/DD4hep/Handle.h#L76-L78